fdbserver/SimKmsConnector.actor.cpp

```diff
@@ -42,19 +43,44 @@
 
 #include "fmt/format.h"
 
+#include <cstring>
 #include <limits>
 #include <memory>
 #include <unordered_map>
 #include <utility>
 
 #include "flow/actorcompiler.h" // This must be the last #include.
 
+#define DEBUG_SIM_KEY_CIPHER DEBUG_ENCRYPT_KEY_CIPHER
+
 using SimEncryptKey = std::string;
 struct SimEncryptKeyCtx {
+	static const int minCipherLen = 4;
+	static const int maxCipherLen = 256;
+
 	EncryptCipherBaseKeyId id;
 	SimEncryptKey key;
+	int keyLen;
+
+	explicit SimEncryptKeyCtx(EncryptCipherBaseKeyId kId, const char* data, const int dataLen)
+	  : id(kId), key(data, dataLen), keyLen(dataLen) {
+		if (DEBUG_SIM_KEY_CIPHER) {
+			TraceEvent(SevDebug, "SimKmsKeyCtxInit").detail("BaseCipherId", kId).detail("BaseCipherLen", dataLen);
+		}
+	}
+
+	static int getKeyLen(const EncryptCipherBaseKeyId id) {
+		ASSERT_GT(id, INVALID_ENCRYPT_CIPHER_KEY_ID);
+
+		int ret = AES_256_KEY_LENGTH;
+		if ((id % 2) == 0) {
+			ret += (id % AES_256_IV_LENGTH);
```

Comment:
Should this be `AES_256_KEY_LENGTH`?

Reply:
Same explanation as the earlier, this method is used to define the baseCipherKeyLen, the min today is 32 bytes.

Reply:
What i mean you used `AES_256_IV_LENGTH` instead of `AES_256_KEY_LENGTH` in this line, was that intentional?

Reply:
Ahh.. nah that is a typo, thanks for pointing it out.